### PR TITLE
Add kubectl-vpa-recommendation plugin to Krew index

### PR DESCRIPTION
**Pull request by wI2L**
Hello

I would like to propose a new plugin I've been working on recently: https://github.com/wI2L/kubectl-commitment
The plugin allows users to compare the target recommendations of `VerticalPodAutoscaler` resources with the actual resource requests of the pods managed by the targeted controller.

Release automation with `rajatjindal/krew-release-bot@v0.0.40` and documentation will be added/updated, if the plugin gets added to the index. The `.krew.yaml` template file is already versioned in the repository: https://github.com/wI2L/kubectl-commitment/blob/master/.krew.yaml

The plugin lacks unit tests/coverage, they will be added over time as the plugin mature in features.

Regards,

William

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

**Comment by ahmetb** [comment]
This seems pretty excellent already. The name "commitment" doesn't really signal to me that this is designed to measure VPA status, but if you feel comfortable with it that's fine as it's specific enough –except the only other use case I can imagine this name could be used is probably for a plugin that measures overcommit/undercommit situation (and this would be a perfect name probably for that plugin instead).
I bring this up since there's still time to change this, and the name didn't really ring the "VPA" bell in my mind.
/approve

File: plugins/commitment.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: commitment
+spec:
+  version: v0.1.1
+  homepage: https://github.com/wI2L/kubectl-commitment
+  shortDescription: Compare VerticalPodAutoscaler recommendations to actual resources requests
```
Comment (ahmetb) [inline plugins/commitment.yaml:8]: this probably won't fit `kubectl krew list` output, it'll be truncated at 50 chars.

Comment (wI2L) [reply plugins/commitment.yaml:8]: Got it. I wrapped the `description` field to 80 chas as indicatd in the developper guide, but missed that. Will fix it.

**Comment by wI2L** [comment]
I am open to suggestions regarding the name of the plugin, if you have any.

> –except the only other use case I can imagine this name could be used is probably for a plugin that measures overcommit/undercommit situation

To me, that's what the plugin is doing, but I may be wrong and mistaking it with another thing. Could you clarify what you have in mind when you mention "undercommit/overcommit" ?

**Comment by ahmetb** [comment]
> Is "commitment, under/over" generally used to talk about limits vs. capacity, instead of requests ?

Yeah, I think typically when people talk about "commitment" in the infrastructure space (compute, network etc), we talk about what's provisioned and what's assigned vs what's available. e.g. memory, CPU, network etc. An alternative to this is "oversubscription" or "undersubscription".

Perhaps the name is okay and I'm exaggerating the potential of someone developing a plugin that shows what's assigned vs what's available in terms of resources like memory/CPU. But Kubernetes doesn't really let us overcommit I guess, so maybe that won't happen.

**Comment by wI2L** [comment]
@ahmetb
Looks like `kubectl describe node` already show the informations regarding allocated resources (and overcommitment when limits > 100%). I have mixed feelings regarding the naming, when it comes to VPA, it seems to me that it's also an over-commitment to go beyond the recommendations (in the sens that this is an excess).
I have thought of alternatives terns, such as `excess`, `reservation` (since it's related to resource requests).
Should we get another opinion ? @corneliusweig maybe ?
Again, not against a renaming of the plugin if we find a name that is clearer.

**Comment by chriskim06** [comment]
`commitment` makes me think of node over commitment like you both discussed earlier. I'm not 100% opposed to that name but I think it would be more clear if `vpa` was included in the name (like `vpa-commitment` or something).

**Comment by wI2L** [comment]
@chriskim06 @ahmetb 
What do you think of `recommendation(s)`, as name ? It's closely relatd to the VPA concept. Ideally I'd like to kep a single-word name, but otherwise `vpa-commitment` sounds good.

File: plugins/commitment.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  description: |
+    This plugin prints a table that show the percentage increase/decrease
+    of the selected VerticalPodAutoscaler recommendations compared to the
+    targeted controller's pod resource requests.
+  caveats: |
+    The plugin recognize only some well-known controllers such as:
```
Comment (chriskim06) [inline plugins/commitment.yaml:14]: small typo `recognize` -> `recognizes`

also I think maybe its worth mentioning that the `VerticalPodAutoscaler` needs to be installed in the cluster, especially depending on if the name will contain `vpa` or not

**Comment by chriskim06** [comment]
> @chriskim06 @ahmetb
> What do you think of `recommendation(s)`, as name ? It's closely relatd to the VPA concept. Ideally I'd like to kep a single-word name, but otherwise `vpa-commitment` sounds good.

actually `vpa-recommendation` sounds even better to me, still can't think of a one word plugin name though. but like I said before, if everyone if fine with `commitment` (or `recommendation`) I'm also ok with it.

File: plugins/commitment.yaml (hunk 2 above)
Comment (wI2L) [reply plugins/commitment.yaml:14]: Yep, make sense. Will push a fix including @ahmetb recommendation to wrap the description to 50 chars.

**Comment by ahmetb** [comment]
Recommendations is way broader IMO and can actually be developed as many different plugins.

**Comment by ahmetb** [comment]
Took a closer look at this plugin. I think `vpa-diff` or `vpa-status` are also pretty good alternatives. As a matter of fact, both you and [this plugin](https://github.com/ninlil/kubectl-vpa) (not in krew) use "diff" terminology commonly. That could be something to think about since your plugin is still new.

I don't see "commitment" as a word within the VPA nomenclature anywhere and that's why the word still does not ring a bell for me in this context.

**Comment by wI2L** [comment]
> I don't see "commitment" as a word within the VPA nomenclature anywhere and that's why the word still does not ring a bell for me in this context.

"commitment" is often used as terminology regarding limits, but that could apply to requests as well, since the requests "commited" are taken into account by the scheduler to schedule pods onto nods (a.k.a reservation). Since the VPA computes targets that indicate whether the resources would be wasted or aren't enough, that's why I went with that from the beginning.

I think we can agree on the fact that the plugin would be better named with a `vpa-` prefix. What completes it is the hard part.

> As a matter of fact, both you and this plugin (not in krew) use "diff" terminology commonly. That could be something to think about since your plugin is still new.

I was aware of this plugin before I developped this one. I wasn't satisfied with the approach of binding pods to VPA by name, rather than fetching the target controller referenced by the VPA. What do you mean by "That could be something to think about since your plugin is still new." ? Should we leave the "vpa-diff" name open if this plugin's owner wish to add his plugin to krew ?

To stick closely to the VPA nomenclature, we could go with the name `vpa-target(s), instead of `vpa-commitment` or `vpa-recommendations`. `vpa-diff` is also ok, but that's not taking into account that a VPA a different "kind/type" of recommendations (target, upper/lower bound, uncapped target).

**Comment by wI2L** [comment]
I gave it a thought again, `vpa-targets` is too specific to a type of recommendation provided by a VPA. Since this plugin will be centered on computing the differences from VPA recommendations vs. actual usage, I think `vpa-recommendation` (singular), would fit nicely. @ahmetb Could you please validate it, before I go on and rename the plugin itlsef, release new binaries and update the PR ? Thanks

**Comment by wI2L** [comment]
Thank you both for the feedback and review. I'll notify you once the plugin is renamed and the PR amended to include name change and review fixes.

**Comment by wI2L** [comment]
@ahmetb I just released a new version of the plugin with the new chosed name, and amended the PR to update the krew manifest file. I think it's ready for another review. Thanks.

**Comment by ahmetb** [comment]
/lgtm
/approve
Thanks!

Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Comment by wI2L** [comment]
Yep, I didn't committed the new step in the `release` pipeline yet, but will do.

**Pull request merged**